pkg/cache/task.go

```diff
@@ -558,3 +560,25 @@ func (task *Task) sanityCheckBeforeScheduling() error {
 	}
 	return nil
 }
+
+func (task *Task) UpdatePodCondition(podCondition *v1.PodCondition) (bool, *v1.Pod) {
+	task.lock.Lock()
+	defer task.lock.Unlock()
+
+	status := task.podStatus.DeepCopy()
+	pod := task.pod.DeepCopy()
```

Comment:
If we have the pod deepcopy, why we also need a deepcopy of podStatus?

Reply:
These are two distinct variables. PodStatus is updated later, but not inside "pod". Otherwise we'd need to obtain locks every time we deepcopy the pod, which is a legit approach, but in this PR I avoided this.

Reply:
Thanks clarify, it makes sense to me now.